### DISCUSSION TITLE
⚠️  Schema must contain uniquely named types but contains multiple types

### DISCUSSION THREAD

**Discussion by tarikrital** [swellstores/community#526] [Help] [answered]
![image](https://github.com/swellstores/community/assets/52907282/c08b7e2b-d4eb-4e0a-add7-e569f13cc8f5)

**Comment by logeshswell** [comment]
@wpplumber When did you start notice this issue?
I could also reproduce this issue in my storefront and confirm it's a bug.

Let me look deeper into this issue and get back to you if I get some updates.

**Reply by tarikrital** [reply]
![image](https://github.com/swellstores/community/assets/52907282/365941cd-998c-48c6-b4ce-7dfc817f84c5)

⌛  This is blocking my client to get the new features I integrated on Horizon theme and me from getting the payment.

**Reply by logeshswell** [reply, marked as answer]
@wpplumber The issue is fixed in your store (raw-foods). Can you check and confirm please?

**Reply by tarikrital** [reply]
Yes, the `graphql-codegen` worked.

**Reply by logeshswell** [reply]
Thanks for the confirmation and very sorry once again for the inconvenience caused to you.

**Reply by logeshswell** [reply]
We are actively working on it to avoid these kind of issues in the future.

**Comment by flonUK** [comment]
Same issue raised on chat. Products are not loading on the site which means customers cannot see or purchase  items for 3 days! I was told this is still being discussed. Don't even know the right emoji to use here.

**Reply by flonUK** [reply]
Day 4!

**Reply by flonUK** [reply]
Please revert whatever caused this change while you deal with it.  The initial plug yesterday does not work and is there a way to track when these events happen? 

I know you might not have ecommerce stores but maybe assume your laptop is dead for 1 hour, now think of that over 4 days? This is a public forum so trying to bite my tongue  and be respectful as I can't imagine the level of work you are doing but if BAU is flawed and customers cannot access products for 4 days running! what is the point of the appstore or anything else and what does it say about customer service?

**Reply by flonUK** [reply]
[FAILED]         GraphQLError: Schema must contain uniquely named types but contains multiple types named "SwellTextLinks".

**Reply by tarikrital** [reply]
Here are some potential reasons why this issue could be originating from the API server:

- Schema generation process: If the GraphQL schema is automatically generated by the API server, there might be a bug or misconfiguration in the schema generation process that results in duplicate type names.

- Schema stitching or federation: If the API server is composed of multiple services or schemas stitched together using tools like Apollo Federation, there might be a conflict in type names between the federated schemas.

- Data model changes: Changes to the underlying data model or schema definition on the server side could inadvertently introduce duplicate type names.

**Reply by logeshswell** [reply]
@flonUK @wpplumber Very sorry for the inconvenience, we are actively working on it and will share an update very shortly today.

**Reply by flonUK** [reply]
@logeshswell working for me too. Thanks . Is there a tracker for these things so I can try to automate it and get an email when down? Thanks
